lib/ngx/pipe.lua

```diff
@@ -566,6 +567,38 @@ do
                     error("bad buffer_size option", 2)
                 end
             end
+
+            if opts.environ then
+                environ = opts.environ
+                local environ_type = type(environ)
+                if environ_type ~= 'table' then
```

Comment:
style: could we stick to double quotes to stay consistent with the rest of this function's body style?